DigitalLearningSolutions.Data/Models/TrackingSystem/DateInformation.cs

```diff
@@ -5,22 +5,61 @@
 
     public class DateInformation
     {
-        public DateInformation(DateTime date, ReportInterval interval)
+        public DateInformation(DateTime startDate, ReportInterval interval)
         {
-            Date = date;
+            StartDate = startDate;
             Interval = interval;
         }
 
         public ReportInterval Interval { get; set; }
-        public DateTime Date { get; set; }
+        public DateTime StartDate { get; set; }
 
         public string GetDateLabel(string format)
         {
-            var quarter = Date.Month / 3 + 1;
+            format = ConvertQuartersInFormatString(format, StartDate);
 
-            format = format.Replace("Q", $"'Quarter' {quarter}").Replace("q", $"Q{quarter}");
+            return StartDate.ToString(format);
+        }
+
+        public string GetDateRangeLabel(string format, DateTime rangeTerminator, bool startRangeFromTerminator)
+        {
+            var dateFromThisObject = startRangeFromTerminator ? GetFinalDate() : StartDate;
+            var formatForDateFromThisObject = ConvertQuartersInFormatString(format, dateFromThisObject);
+
+            var formatForTerminator = ConvertQuartersInFormatString(format, rangeTerminator);

```

Comment:
I find these variable/parameter names a bit confusing, particularly formatForDateFromThisObject. Isn't this actually a formatted date not a format?

Reply:
It's a format string for the .ToString() calls below. The reason it also contains formatted date elements is that C#'s datetime formatting doesn't have any handling for Q/quarters, so it processes them there and then passes the format string on to ToString(). The already-processed quarter elements aren't interpreted by the ToString() date formatter so are passed through unchanged.